### ISSUE TITLE
Doesn't recognize BASH_ARGV0

### ISSUE THREAD

**Issue by sergio-costas** [koalaman/shellcheck#2554]
I'm assigning in a bash script the variable BASH_ARGV0, which is the way of modifying the value of $0. Unfortunately, checkshell doesn't recognize it and shows an SC2034 UNUSED VARIABLE. Tested online and with version 0.8.0.

Example code:

```
#!/bin/bash

# emulate "exec $@" with "source"
BASH_ARGV0=$1
shift
# shellcheck source=/dev/null
source "$0"
```

- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

**Comment by larryv** [comment]
This should be fixed by 966fb3e3dd3b49e64f022478f18f7aa4bad53a27.

**Comment by sergio-costas** [comment]
Thanks!!!!
